lr-cli/cortx_setup/commands/node/finalize.py

```diff
@@ -29,9 +34,86 @@ class NodeFinalize(Command):
+        resource_paths = [HEALTH_PATH, MANIFEST_PATH]
+        for resource in resource_paths:
+            path = get_pillar_data(resource)
+            if not path or path is MISSED:
+                raise CortxSetupError(f"{resource} resource is not configured")
+            if not Path(path).is_file():
+                raise CortxSetupError(f"Validation failed: "
+                                      f"File not present {path}")
+        self.logger.debug("Node health check: Success")
+
+    def _validate_cert_installation(self):
+        self.logger.debug("Validating certificate installtion check")
+        cert_file = None
+        cert_file = Path(CERT_PATH / 'stx.pem')
+        if not cert_file.is_file():
+            raise CortxSetupError(
+                f"Validation failed: "
+                f"Cert file not present {cert_file}"
+            )
+        self.logger.debug("Certificate installtion check: Success")
+
+    def _validate_interfaces(self, minion_id):
+        self.logger.debug("Validating network interfaces check")
+        mgmt = get_pillar_data(f'cluster/{minion_id}/network/mgmt/interfaces')
+        private_data = get_pillar_data(f'cluster/{minion_id}/network/data/private_interfaces')  # noqa: E501
+        public_data = get_pillar_data(f'cluster/{minion_id}/network/data/public_interfaces')  # noqa: E501
+        if not mgmt or mgmt is MISSED:
+            raise CortxSetupError(f"Mgmt interfaces are not provided")
+        if not private_data or private_data is MISSED:
+            raise CortxSetupError(f"Private data interfaces are not provided")
+        if not public_data or public_data is MISSED:
+            raise CortxSetupError(f"Public data interfaces are not provided")
+
+        for interface in mgmt:
+            if interface in private_data or interface in public_data:
+                raise CortxSetupError(
+                    f"Same interface provided for mgmt and data")
+
+        for interface in private_data:
+            if interface in public_data:
+                raise CortxSetupError(
+                    f"Same interface provided for public_data "
+                    f"{public_data}& private_data: {private_data}"
+                )
+        interfaces(mgmt + private_data + public_data)
+        self.logger.debug("Network interfaces check: Success")
+
+    def _validate_devices(self, minion_id, s_type):
+        self.logger.debug("Validating cvg devices check")
+        cvgs = get_pillar_data(f'cluster/{minion_id}/storage/cvg')
+        if not cvgs or cvgs is MISSED:
+            raise CortxSetupError(f"Devices are not provided")
+        for cvg in cvgs:
+            meta_data = cvg.get('metadata_devices')
+            data = cvg.get('data_devices')
+            if not meta_data or not data:
+                raise CortxSetupError(f"metadata or data devices are missing")
+
+            for meta in meta_data:
+                if meta in data:
```

Comment:
Do we not need similar check check for data in meta_data?

Reply:
I think only checking meta in data will raise error for duplicate devices vice versa is not required.